src/dom/domBooking.js

```diff
@@ -1,10 +1,164 @@
+  if (e.target.classList.contains("filter-modal-open")) {
+    filterModal.showModal();
+  } else if (e.target.classList.contains("filter-modal-close")) {
+    filterModal.close();
+  } else if (e.target.classList.contains("book-room-button")) {
+    const calendar = document.querySelector(".booking-date");
+
+    const roomDOM = e.target.closest(".room-card");
+    const roomNumber = +roomDOM.dataset.number;
+    const room = getRoom(roomNumber, localData.getRooms());
+
+    // fixes the bug of calendar input being in local time
+    // but Date uses UTC time and so they will clash and change day back
+    const calendarParse = new Date(`${calendar.value}T00:00:00`);
+
+    const booking = createBooking(
+      currentCustomer.id,
+      calendarParse,
+      room.number
+    );
+
+    addBooking(localData, booking).then(() =>
+      setDOM(document.getElementById("root"), () => bookingPage(calendarParse))
+    );
+  }
+});
+
+document.getElementById("root").addEventListener("change", (e) => {
+  if (e.target.classList.contains("booking-date")) {
+    const calendar = document.querySelector(".booking-date");
+    const calendarParse = new Date(`${calendar.value}T00:00:00`);
+    setDOM(e.currentTarget, () => bookingPage(calendarParse));
+  }
+});
+
+export function bookingPage(date = new Date(startOfToday())) {
   const anchor = document.createElement("div");
   anchor.innerHTML = `
   <h1>Booking Screen</h1>
   <button class="history-button">View history</button>
   <button class="logoff-button">Log off</button>
-  `;
+  
+  <hr>
+  
+  <input 
+    type="date" 
+    class="booking-date" 
+    value="${format(date, "yyyy-MM-dd")}">
+
+  <button>Clear filters</button>
+  <div class="dropdown">
+    <button class="dropbtn">Price</button>
+    <div class="dropdown-content">${priceFilterHTML()}</div>
+  </div>
+  <div class="dropdown">
+    <button class="dropbtn">Bed #</button>
+    <div class="dropdown-content">${bedNumberFilterHTML()}</div>
+  </div>
+  <div class="dropdown">
+    <button class="dropbtn">Room Type</button>
+    <div class="dropdown-content">${roomTypeFilterHTML()}</div>
+  </div>
+  <div class="dropdown">
+    <button class="dropbtn">Bed Size</button>
+    <div class="dropdown-content">${bedSizeFilterHTML()}</div>
+  </div>
+
+  <hr>
+
+  ${availableRoomsHTML(localData, date)}`;
 
   return anchor;
 }
+
+function availableRoomsHTML(data, date) {
+  const rooms = getAvailableRooms(data, date);
+  return rooms.reduce((html, room) => {
+    html += roomCardHTML(room, date);
+    return html;
+  }, "");
+}
+
+function roomCardHTML(room, date) {
+  return `
+  <section class="room-card" data-number="${room.number}">
+    <div class="booking">Room ${room.number}</div>
+    <div>Room type: ${room.roomType}</div>
+    <div>Has bidet: ${room.bidet}</div>
+    <div>Bed size: ${room.bedSize}</div>
+    <div>Bed #: ${room.numBeds}</div>
+    <div>Price ${room.costPerNight}</div>
+    ${bookButtonHTML(date)}
+  </section>`;
```

Comment:
Curious about the choice of divs vs ul since it feels like the items under booking are a 'list of attributes of the room listed on the card. I assume using ul caused some kind of problem, just curious what that was.

Reply:
I didn't focus on semantic HTML or accessibility during this iteration and just worked on getting the code functioning and fixing bugs. This specific instance has been fixed to use <ul> now instead of <div>